src/server/game/Server/WorldSession.cpp

```diff
@@ -847,8 +847,20 @@ void WorldSession::ReadMovementInfo(WorldPacket &data, MovementInfo* mi)
         in conjunction with any of the moving movement flags such as MOVEMENTFLAG_FORWARD.
         It will freeze clients that receive this player's movement info.
     */
-    REMOVE_VIOLATING_FLAGS(mi->HasMovementFlag(MOVEMENTFLAG_ROOT),
-        MOVEMENTFLAG_ROOT);
+    if (!GetPlayer()->GetVehicleBase())
+    {
+        // Only remove here for non vehicles
+        REMOVE_VIOLATING_FLAGS(mi->HasMovementFlag(MOVEMENTFLAG_ROOT),
+            MOVEMENTFLAG_ROOT);
+    }
+    else
+    {
+        // Actually players in rooted vehicles still send commands, don't clear root for these!
+        // Check for non movement flag. Otherwise, also clear for vehicles
+        if (!GetPlayer()->GetVehicleBase()->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE))
+            REMOVE_VIOLATING_FLAGS(mi->HasMovementFlag(MOVEMENTFLAG_ROOT),
+                MOVEMENTFLAG_ROOT);
```

Comment:
This once again opens up the exploit of sending fake movement packets with root+movement flags to freeze all nearby players (but only if player is on a rooted vehicle)